src/OperatorMsg.chpl

```diff
@@ -71,309 +71,146 @@ module OperatorMsg
+            return doBinOpvv(l, r, real, op, pn, st);
+          } else {
+            // isn't +, -, /, // so we can use LHS to determine type
+            return doBinOpvv(l, r, uint, op, pn, st);
           }
-          when (DType.UInt64, DType.BigInt) {
-            var l = toSymEntry(left,uint, nd);
-            var r = toSymEntry(right,bigint, nd);
-            if boolOps.contains(op) {
-              // call bigint specific func which returns distr bool array
-              var e = st.addEntry(rname, createSymEntry(doBigIntBinOpvvBoolReturn(l, r, op)));
-              var repMsg = "created %s".format(st.attrib(rname));
-              return new MsgTuple(repMsg, MsgType.NORMAL);
-            }
-            // call bigint specific func which returns dist bigint array
-            var (tmp, max_bits) = doBigIntBinOpvv(l, r, op);
-            var e = st.addEntry(rname, createSymEntry(tmp, max_bits));
-            var repMsg = "created %s".format(st.attrib(rname));
-            return new MsgTuple(repMsg, MsgType.NORMAL);
+        }
+        else if binop_dtype_a == int && binop_dtype_b == uint {
+          if boolOps.contains(op) {
+            return doBinOpvv(l, r, bool, op, pn, st);
           }
-          when (DType.Bool, DType.BigInt) {
-            var l = toSymEntry(left,bool, nd);
-            var r = toSymEntry(right,bigint, nd);
-            if boolOps.contains(op) {
-              // call bigint specific func which returns distr bool array
-              var e = st.addEntry(rname, createSymEntry(doBigIntBinOpvvBoolReturn(l, r, op)));
-              var repMsg = "created %s".format(st.attrib(rname));
-              return new MsgTuple(repMsg, MsgType.NORMAL);
-            }
-            // call bigint specific func which returns dist bigint array
-            var (tmp, max_bits) = doBigIntBinOpvv(l, r, op);
-            var e = st.addEntry(rname, createSymEntry(tmp, max_bits));
-            var repMsg = "created %s".format(st.attrib(rname));
-            return new MsgTuple(repMsg, MsgType.NORMAL);
+          // +, -, /, // both result in real outputs to match NumPy
+          if realOps.contains(op) {
+            return doBinOpvv(l, r, real, op, pn, st);
+          } else {
+            // isn't +, -, /, // so we can use LHS to determine type
+            return doBinOpvv(l, r, int, op, pn, st);
           }
         }
-        var errorMsg = unrecognizedTypeError(pn, "("+dtype2str(left.dtype)+","+dtype2str(right.dtype)+")");
-        omLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-        return new MsgTuple(errorMsg, MsgType.ERROR);
+        else if (binop_dtype_a == bigint || binop_dtype_b == bigint) &&
+                !isRealType(binop_dtype_a) && !isRealType(binop_dtype_b)
+        {
+          if boolOps.contains(op) {
+            // call bigint specific func which returns distr bool array
+            return st.insert(new shared SymEntry(doBigIntBinOpvvBoolReturn(l, r, op)));
```

Comment:
`doBigIntBinOpvvBoolReturn` is a special method that handles the few bigint operations that return an array of `bool`. I'm guessing it was factored out into its own helper function because all the other bigint ops return an array of `bigint`.